specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json

```diff
@@ -5068,6 +5089,7 @@
           "description": "The list of tag values."
         }
       },
+      "x-ms-azure-resource": true,
```

Comment:
>"x-ms-azure-resource": true, [](start = 6, length = 28)

why do we need this?

Reply:
Running autorest validation
"autorest --azure-validator --input-file=resources.json --openapi-type=arm" gave out few errors. Tried to resolve as much as possible. 

Error related to the above change was 
ERROR (XmsResourceInPutResponse/R2062/ARMViolation): The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: 'Tags_CreateOrUpdate' Model: 'TagDetails'.

If this is not mandatory, I could skip this.

Reply:
Validation complains if this is missing for an ARM response

Reply:
but what does this imply? Is TagDetails an "ARM resource"?

Reply:
TagDetails is in this format as well 
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-azure-resource
The same has been marked for TagValue and TagsResource

Reply:
`TagDetails` gets returned by one of our APIs `"/subscriptions/{subscriptionId}/tagNames/{tagName}"`. The validation complains, and this is the right fix for it. However, this is not related to the rest of this PR.

Reply:
It is not necessarily an ARM resource. Somewhere in our specs we have said that anything ARM returns as a response to an API call, should have this.

Reply:
I see. thanks for the details. I've approved the PR.